torch/distributed/_tensor/_utils.py

```diff
@@ -21,6 +21,7 @@ def compute_local_shape(
     Compute the shape of a local shard of the given DTensor on its current
     coordinate of the mesh.
     """
+    # TODO(xilun): address new Shard type
```

Comment:
hmmm would local shape really matters for strided shard placement? I think for either shard/strideShard they should produce the same local shape, except for the case when it works with uneven sharding? Let's be more clear what the TODO mean or simply remove this

Reply:
You're right, strided sharding only differs on where the shard is placed and it doesn't affect shape but only offset. 

For uneven padding, I think the above claim still holds true because of SPMD -- every shard will have the same shape after padding.